modules/AOP_Case_Updates/AOP_Case_Updates.php

```diff
@@ -134,6 +134,24 @@ public function save($check_notify = false)
         return $this->id;
     }
 
+    /**
+     * Fixes unclosed HTML tags
+     */
+    private function parseDescription()
+    {
+        $description = SugarCleaner::cleanHtml($this->description);
+
+        if (preg_match('/<[^<]+>/', $description, $matches) !== 0) {
+            $doc = new DOMDocument();
+            $doc->loadHTML($description);
+            $doc->removeChild($doc->doctype);
+            $doc->replaceChild($doc->firstChild->firstChild->firstChild, $doc->firstChild);
```

Comment:
can you use the DOMDocument::getElementsByTagName() for find the `<body>` and other tags or you also can use `$dom->validate()`
https://stackoverflow.com/questions/5030392/x-html-validator-in-php